TypeScript compilation fails on fresh install following README instructions

**Description:**

Fresh installation following README.md instructions fails at `npm run build` with TypeScript errors in WhatsApp session handling.

**Reproduction Steps:**

```bash
git clone https://github.com/letta-ai/lettabot.git
cd lettabot
npm install
npm run build
```

**Error Output:**

```
src/channels/whatsapp/session.ts:195:16 - error TS2349: This expression is not callable.
  Type 'typeof import("/home/jigonr/lettabot/node_modules/@whiskeysockets/baileys/lib/index")' has no call signatures.

195   const sock = makeWASocket({
                   ~~~~~~~~~~~~

src/channels/whatsapp/session.ts:207:24 - error TS7006: Parameter 'key' implicitly has an 'any' type.

207     getMessage: async (key) => {
                           ~~~
```

**Environment:**
- OS: Linux (Ubuntu/Debian)
- Package manager: npm (also tested with bun, same issue)
- Node version: [your version]

**Expected Behavior:**
Following README.md setup instructions should result in successful compilation.

**Actual Behavior:**
TypeScript compilation fails, preventing users from completing basic setup.

**Suggested Fixes:**
1. Fix `makeWASocket` import in `src/channels/whatsapp/session.ts`
2. Add type annotation for `key` parameter (line 207)
3. Add pre-commit hooks or CI checks to prevent broken commits from landing in main
4. Test that `npm install && npm run build` succeeds before releases

**Impact:**
New users cannot complete installation following official documentation.

--- a/package.json
+++ b/package.json
@@ -65,7 +65,7 @@
   },
   "optionalDependencies": {
     "@slack/bolt": "^4.6.0",
-    "@whiskeysockets/baileys": "^6.7.21",
+    "@whiskeysockets/baileys": "6.7.21",
     "discord.js": "^14.25.1"
   },
   "devDependencies": {

--- a/src/channels/whatsapp/session.ts
+++ b/src/channels/whatsapp/session.ts
@@ -204,7 +204,7 @@ export async function createWaSocket(options: SocketOptions): Promise<SocketResu
     logger: logger as any,
     printQRInTerminal: false,
     // getMessage for retry capability - store is populated when we SEND messages, not here
-    getMessage: async (key) => {
+    getMessage: async (key: { id?: string | null }) => {
       if (!key.id) return undefined;
       return messageStore.get(key.id);
     },

--- a/src/core/commands.test.ts
+++ b/src/core/commands.test.ts
@@ -66,10 +66,11 @@ describe('COMMANDS', () => {
     expect(COMMANDS).toContain('heartbeat');
     expect(COMMANDS).toContain('help');
     expect(COMMANDS).toContain('start');
+    expect(COMMANDS).toContain('reset');
   });
 
-  it('has exactly 4 commands', () => {
-    expect(COMMANDS).toHaveLength(4);
+  it('has exactly 5 commands', () => {
+    expect(COMMANDS).toHaveLength(5);
   });
 });
 

